threadregular.go

```diff
@@ -99,39 +111,36 @@ func (handler *regularThread) afterRequest() {
 
 func handleRequestWithRegularPHPThreads(ch contextHolder) error {
 	metrics.StartRequest()
+	metrics.QueuedRequest()
 
-	select {
-	case regularRequestChan <- ch:
-		// a thread was available to handle the request immediately
-		<-ch.frankenPHPContext.done
+	if err := acquireSemaphoreWithAdmissionControl(regularSemaphore, scaleChan, ch.frankenPHPContext); err != nil {
```

Comment:
You should pass the context from `ch.ctx` to `acquireSemaphoreWithAdmissionControl` and use it where appropriate (this will allow better shutdown and reload when using Caddy, for instance).

Reply:
I'll give it a go and see if I can find a context we can pass through without breaking tests.